theano/gpuarray/dnn.py

```diff
@@ -2833,6 +2833,295 @@ def local_abstractconv3d_cudnn_graph(op, context_name, inputs, outputs):
     return [rval]
 
 
+class GpuDnnTransformerDescriptor(COp):
+
+    """
+    This Op builds a spatial transformer descriptor for use in spatial transformer network
+    operations.
+    """
+
+    __props__ = ('dtype',)
+    params_type = ParamsType(dtype=cudnn.cudnnDataType_t)
+
+    def c_headers(self):
+        return ['cudnn.h', 'cudnn_helper.h']
+
+    def c_header_dirs(self):
+        header_dirs = [os.path.dirname(__file__)]
+        if config.dnn.include_path:
+            header_dirs += [config.dnn.include_path]
+        return header_dirs
+
+    def c_libraries(self):
+        return ['cudnn']
+
+    def c_lib_dirs(self):
+        lib_dirs = []
+        if config.dnn.library_path:
+            lib_dirs += [config.dnn.library_path]
+        return lib_dirs
+
+    def do_constant_folding(self, node):
+        return False
+
+    def __init__(self, dtype=theano.config.floatX):
+        COp.__init__(self, ["c_code/dnn_sptf_desc.c"], "APPLY_SPECIFIC(dnn_sptf_desc)")
+        assert cudnn.cudnnDataType_t.has_alias(dtype)
+        self.dtype = dtype
+
+    def make_node(self, dimensions):
+        dimensions = as_tensor_variable(dimensions)
```

Comment:
You haven't checked `dimensions` dtype, but in C code you assume that it's an integer array (as you use `(int*) PyArray_GETPTR1(...)`).

What I suggest is:
- check that `dimensions` ndim is 1 (we expect a vector).
- check that `dimensions` dtype is an integer type (dtype should be in `theano.tensor.basic.int_dtypes`).
- apply as cast to force dimensions to have a specific type in C code (this will let the user use any integer type he wants). I suggest you use int64 as it is the biggest supported integer type. Example: `dimensions = theano.tensor.basic.cast(dimensions, 'int64')`
- Then in C code use `(npy_int64*)` instead of `(int*)`. Also, don't forget that an `int` in C code does not have a standard size accross all systems and all compilers, so be careful when you can something to an int !